### PR TITLE
mc_pos_control: adjust thrust in xy when altitude and position hold is engaged

### PR DESCRIPTION
**Pull request by Stifael**
turn off when pos and alt hold engaged and thrust z is low

### CODE REVIEW

**Comment by Stifael** [comment]
This PR follows #6434. 
here is a silt plot:
http://logs.px4.io/plot_app?log=700d5cbd-21d0-4718-a222-d8ad11a8a534
this PR improves the current upward flight, but there is still a roll setpoint request apparent.

**Comment by Stifael** [comment]
I still would add acceleration limits to make transitions smoother

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 1]
```diff
@@ -1890,13 +1884,34 @@ MulticopterPositionControl::control_position(float dt)
...
-		float thrust_body_z = F.dot(-R_z); /* recalculate because it might have changed */
+		matrix::Vector3f R_z(_R(0, 2), _R(1, 2), _R(2, 2));
+		float thrust_body_z = F.dot(-R_z);
+		thrust_sp = thrust_sp.normalized() * fabsf(thrust_body_z);
+
+		/* To keep the attitude at low thrust_setpoint_z, we ajdust the thrust_setpoint_xy such that no
+		 * jerk happens when entering altitude hold when position hold is engaged
+		 */
+		bool keep_attitude = _pos_hold_engaged && _alt_hold_engaged && (-thrust_sp(2) < 2.0f * _params.thr_min);
+		if(keep_attitude){
```
Comment (LorenzMeier) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: Code style

**Comment by LorenzMeier** [comment]
Implementation looks good. I guess we need to look at the performance and side effects to see if it needs additional work. Great job!

**Comment by LorenzMeier** [comment]
Jenkins test this please.

**Comment by LorenzMeier** [comment]
@Stifael Code style fail. Please let Julian or Beat show you how to install the pre-commit hook so you can catch this locally.

**Comment by Stifael** [comment]
This log is from flight with thrust_threshold_z to thrust_hover :
http://logs.uaventure.com/view/4Lg3h4ZUJXpYA49NJQ6GbK
This log is from flight with thrust_threshold_z to 2.0 thrust_min
http://logs.uaventure.com/view/tjCrSs83wgKAxMynjhozUA

With thrust_min, the jerk was still apparent. With thrust_hover, the jerk was almost gone. However,setting the thrust_threshold to hover too high because the attitude gets adjusted even in situation where it is not required. One possibility is to make the thrust_threshold to some percentage of the thrust_hover, but that assumes that the thrust_hover is set close to the actual thrust_hover.

In addition an acceleration limit in z-axis is still beneficial. I will also add that

**Comment by Stifael** [comment]
I close this in favor of #6491 since it is simpler and has no restriction on the flight performance.

**Pull request closed without merging**